packages/nodes-base/nodes/HomeAssistant/HomeAssistant.node.ts

```diff
@@ -0,0 +1,395 @@
+						const additionalFields = this.getNodeParameter('additionalFields', i) as IDataObject;
+
+						const body = {
+						};
+
+						if (Object.entries(additionalFields).length !== 0) {
+							const eventAttributesUi = ((additionalFields!.attributes as IDataObject).eventAttributesUi as IDataObject);
+							if (eventAttributesUi.attribute !== undefined) {
+								(eventAttributesUi.attribute as IDataObject[]).map(
+									attribute => {
+										// @ts-ignore
+										body[ attribute.name as string ] = attribute.value;
+									},
+								);
+							}
+						}
+
+						responseData = await homeAssistantApiRequest.call(this, 'POST', `/events/${eventType}`, body);
+
+					}
+				} else if (resource === 'log') {
+					if (operation === 'getErroLogs') {
+						responseData = await homeAssistantApiRequest.call(this, 'GET', '/error_log');
+						if (responseData) {
+							responseData = {
+								errorLog: responseData,
+							};
+						}
+					} else if (operation === 'getLogbookEntries') {
+						const additionalFields = this.getNodeParameter('additionalFields', i) as IDataObject;
+						let endpoint = '/logbook';
+
+						if (Object.entries(additionalFields).length !== 0) {
+							if (additionalFields.startTime) {
+								endpoint = `/logbook/${additionalFields.startTime}`;
+							}
+							if (additionalFields.endTime) {
+								qs.end_time = additionalFields.endTime;
+							}
+							if (additionalFields.entityId) {
+								qs.entity = additionalFields.entityId;
+							}
+						}
+
+						responseData = await homeAssistantApiRequest.call(this, 'GET', endpoint, {}, qs);
+
+					}
+				} else if (resource === 'template') {
+					if (operation === 'create') {
+						const body = {
+							template: this.getNodeParameter('template', i) as string,
+						};
+						responseData = await homeAssistantApiRequest.call(this, 'POST', '/template', body);
+						if (responseData) {
+							responseData = { renderedTemplate: responseData };
+						}
+					}
+				} else if (resource === 'history') {
+					if (operation === 'getAll') {
+						const returnAll = this.getNodeParameter('returnAll', i) as boolean;
+						const additionalFields = this.getNodeParameter('additionalFields', i) as IDataObject;
+						let endpoint = '/history/period';
+
+						if (Object.entries(additionalFields).length !== 0) {
+							if (additionalFields.startTime) {
+								endpoint = `/history/period/${additionalFields.startTime}`;
+							}
+							if (additionalFields.endTime) {
+								qs.end_time = additionalFields.endTime;
+							}
+							if (additionalFields.entityIds) {
+								qs.filter_entity_id = additionalFields.entityIds;
+							}
+							if (additionalFields.minimalResponse === true) {
+								qs.minimal_response = additionalFields.minimalResponse;
+							}
+							if (additionalFields.significantChangesOnly === true) {
+								qs.significant_changes_only = additionalFields.significantChangesOnly;
+							}
+						}
```

Comment:
Another option would be to use snake case in the param names so that here you can use `Object.assign`.

Reply:
That will make the code shorter but it will not be consistant with the node coding practices